src/main.rs

```diff
@@ -119,8 +114,12 @@ fn main() {
         args.max_blob_size,
         args.log_level,
     );
-    if !args.no_tls {
-        builder.with_tls(args.cert, args.key);
+    if let Some(tls) = args.tls {
+        if tls.len() != 2 {
+            eprintln!("tls must be a pair of paths, cert then key (got: {tls:?})");
+            std::process::exit(1);
+        }
+        builder.with_tls(tls[0].clone(), tls[1].clone());
```

Comment:
Rust newbie here, but why is `clone` needed, as those values are not used anymore in the current function?

Reply:
The issue is that calling `tls[0]` consumes `tls` (because `tls` is a vec of `String`, which does not implement `Copy`). So then I can't call `tls[1]` anymore :/
I'm not aware of any `let (cert, key) = tls;` construct either (without [external dependencies](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.tuples)).